xbmc/dialogs/GUIDialogSmartPlaylistEditor.cpp

```diff
@@ -420,73 +475,20 @@ void CGUIDialogSmartPlaylistEditor::UpdateRuleControlButtons()
                               m_playlist.m_ruleCombination.m_rules[iItem]->m_field != FieldNone); // and it is not be empty
 }
 
-void CGUIDialogSmartPlaylistEditor::OnWindowLoaded()
-{
-  CGUIDialog::OnWindowLoaded();
-
-  SendMessage(GUI_MSG_SET_TYPE, CONTROL_NAME, 0, 16012);
-  // setup the match spinner
-  std::vector< std::pair<std::string, int> > labels;
-  labels.push_back(make_pair(g_localizeStrings.Get(21425), 0));
-  labels.push_back(make_pair(g_localizeStrings.Get(21426), 1));
-  SET_CONTROL_LABELS(CONTROL_MATCH, m_playlist.m_ruleCombination.GetType() == CSmartPlaylistRuleCombination::CombinationAnd ? 0 : 1, &labels);
-
-  // and now the limit spinner
-  labels.clear();
-  labels.push_back(make_pair(g_localizeStrings.Get(21428), 0));
-  const int limits[] = { 10, 25, 50, 100, 250, 500, 1000 };
-  for (unsigned int i = 0; i < sizeof(limits) / sizeof(int); i++)
-    labels.push_back(make_pair(StringUtils::Format(g_localizeStrings.Get(21436).c_str(), limits[i]), limits[i]));
-  SET_CONTROL_LABELS(CONTROL_LIMIT, 0, &labels);
-}
-
 void CGUIDialogSmartPlaylistEditor::OnInitWindow()
 {
   m_cancelled = false;
 
-  SendMessage(GUI_MSG_ITEM_SELECT, CONTROL_LIMIT, m_playlist.m_limit);
-
-  std::vector<PLAYLIST_TYPE> allowedTypes;
-  if (m_mode == "partymusic")
-  {
-    allowedTypes.push_back(TYPE_SONGS);
-    allowedTypes.push_back(TYPE_MIXED);
-  }
-  else if (m_mode == "partyvideo")
-  {
-    allowedTypes.push_back(TYPE_MUSICVIDEOS);
-    allowedTypes.push_back(TYPE_MIXED);
-  }
-  else if (m_mode == "music")
-  { // music types + mixed
-    allowedTypes.push_back(TYPE_SONGS);
-    allowedTypes.push_back(TYPE_ALBUMS);
-    allowedTypes.push_back(TYPE_ARTISTS);
-    allowedTypes.push_back(TYPE_MIXED);
-  }
-  else if (m_mode == "video")
-  { // general category for videos
-    allowedTypes.push_back(TYPE_MOVIES);
-    allowedTypes.push_back(TYPE_TVSHOWS);
-    allowedTypes.push_back(TYPE_EPISODES);
-    allowedTypes.push_back(TYPE_MUSICVIDEOS);
-    allowedTypes.push_back(TYPE_MIXED);
-  }
-  // add to the spinner
-  std::vector< std::pair<std::string, int> > labels;
-  for (unsigned int i = 0; i < allowedTypes.size(); i++)
-    labels.push_back(make_pair(GetLocalizedType(allowedTypes[i]), allowedTypes[i]));
-  // check our playlist type is allowed
+  std::vector<PLAYLIST_TYPE> allowedTypes = GetAllowedTypes(m_mode);
+  // check if our playlist type is allowed
   PLAYLIST_TYPE type = ConvertType(m_playlist.GetType());
   bool allowed = false;
-  for (unsigned int i = 0; i < allowedTypes.size(); i++)
-    if (type == allowedTypes[i])
+  for (auto allowedType: allowedTypes)
```

Comment:
Please use curly braces (even though the original code was missing them as well).